src/components/ActivityIndicatorView.tsx

```diff
@@ -23,23 +23,26 @@ export function ActivityIndicatorView({
   const [showMessage] = useTimeout(timeOutMilliseconds || 5000);
   const { CustomActivityIndicatorView } = useDeveloperConfig();
 
-  return (
-    CustomActivityIndicatorView ?? (
-      <View style={styles.view}>
-        <ActivityIndicator
-          size="large"
-          animating
-          testID={tID('activity-indicator-view')}
-          color={colors.primarySource}
-          {...props}
-        />
-        {timeOutMessage && showMessage() && (
-          <Text variant="labelSmall" style={styles.text}>
-            {timeOutMessage}
-          </Text>
-        )}
-      </View>
-    )
+  const timeOutMessage =
+    !!messageText && !!showMessage() ? (
```

Comment:
I think I initially missed this: I saw that the useTimeout hook causes a component to re-render after the timeout. Does that just make this component show the message after X amount of time?

Reply:
Yeah it does. We had issues early on where we would just see a loader and have no idea why it was spinning forever. Now a message pops up after some time which aids in debugging.